test/unit/collector/serverless.test.js

```diff
@@ -175,11 +287,12 @@ describe('ServerlessCollector API', () => {
     it('compresses full payload and writes formatted to stdout', (done) => {
       api.payload = {type: 'test payload'}
       api.flushPayload(() => {
-        const logPayload = JSON.parse(logStub.args[0][0])
+        const logPayload = JSON.parse(logStub.args[0][1])
```

Comment:
I see the argument here is changing because `writeFileSync` is now being stubbed, where the payload is the second argument.

It is a bit confusing with reusing `logStub`. I'd prob rename to `writeFileSyncStub`. While it would require further renaming if we change the mechanism, I think the clarity of understanding the pieces in play would be worth it.

Reply:
@michaelgoin I think the diff is confusing here--this is within the new `ServerlessCollector with constructor-injected pipe`. 

I'm doing some branching in the original logStub, though, that I'll take out for clarity. The pipe writing is tested here, so that should be enough.

Reply:
i guess i'm trying to determine why we'd need to index [1] instead of [0]. i assumed that was because `logStub` is not actually a log stub anymore but a `writeFileSync` stub, which has different positional arguments.

Reply:
Yes--there's a different argument position. (I've updated the names of the stub and payload vars/consts to reflect this.)